social_engagement/handlers.py

```diff
@@ -11,107 +9,146 @@
     thread_id = getattr(comment, 'thread_id', None)
     parent_id = getattr(comment, 'parent_id', None)
     action_user = kwargs['user']
-    social_engagement_enabled = settings.FEATURES.get('ENABLE_SOCIAL_ENGAGEMENT', False)
 
-    if social_engagement_enabled and course_id and action_user:
-        task_update_user_engagement_score.delay(unicode(course_id), action_user.id)
+    if action_user:
+        # a comment is a reply to a thread
+        # a response is a reply to a response
 
-        # a response is a reply to a thread
-        # a comment is a reply to a response
-        is_comment = not thread_id and parent_id
+        # It's a comment
+        if not parent_id:
+            _increment(action_user.id, course_id, 'num_comments')
 
-        if is_comment:
-            # If creating a comment, then we don't have the original thread_id
-            # so we have to get it from the parent
-            comment = cc.Comment.find(parent_id)
-            thread_id = comment.thread_id
-            replying_to_id = comment.user_id
-
-            # update the engagement of the author of the response
-            task_update_user_engagement_score.delay(unicode(course_id), replying_to_id)
+        # It's a reply
+        else:
+            _increment(action_user.id, course_id, 'num_replies')
 
         if thread_id:
             thread = cc.Thread.find(thread_id)
 
             # IMPORTANT: we have to use getattr here as
             # otherwise the property will not get fetched
             # from cs_comment_service
-            thread_user_id = int(getattr(thread, 'user_id', 0))
+            thread_user_id = int(getattr(thread, 'user_id', None))
 
-            # update the engagement score of the thread creator
-            # as well
-            task_update_user_engagement_score.delay(unicode(course_id), thread_user_id)
+            # update the engagement score of the thread creator as well
+            _increment(thread_user_id, course_id, 'num_comments_generated')
 
 
 @receiver(thread_followed)
+def thread_follow_signal_handler(sender, **kwargs):  # pylint: disable=unused-argument
+    """
+    Updates user social engagement score
+    """
+    thread = kwargs['post']
+    course_id = getattr(thread, 'course_id', None)
+    user_id = getattr(thread, 'user_id', None)
+    action_user = kwargs['user']  # user who followed or un-followed thread
+
+    if user_id and action_user.id != int(user_id):
+        _increment(user_id, course_id, 'num_thread_followers')
+
+
 @receiver(thread_unfollowed)
-def thread_follow_unfollow_signal_handler(sender, **kwargs):  # pylint: disable=unused-argument
+def thread_unfollow_signal_handler(sender, **kwargs):  # pylint: disable=unused-argument
     """
     Updates user social engagement score
     """
     thread = kwargs['post']
     course_id = getattr(thread, 'course_id', None)
     user_id = getattr(thread, 'user_id', None)
     action_user = kwargs['user']  # user who followed or un-followed thread
```

Comment:
Nit: since this is for unfollowing, we can just say `user who un-followed thread`.